client/eth/connection_pool.go

```diff
@@ -36,6 +36,17 @@ func NewConnectionPoolImpl(cfg ConnectionPoolConfig, logger log.Logger) (Connect
 		cfg.HealthCheckInterval = defaultHealthCheckInterval
 	}
 
+	// The LRU cache needs at least one URL provided for both HTTP and WS.
+	hasEthHttpUrls := len(cfg.EthHTTPURLs) > 0
+	if !hasEthHttpUrls {
```

Comment:
```suggestion
	if len(cfg.EthHTTPURLs) == 0 {
```
A new variable defn isn't needed here, as the single line if-check implies the meaning well enough